chain/gen/gen.go

```diff
@@ -399,23 +444,130 @@ func IsRoundWinner(ctx context.Context, ts *types.TipSet, round int64, miner add
 		return false, nil, xerrors.Errorf("failed to get miner worker: %w", err)
 	}
 
-	vrfout, err := ComputeVRF(ctx, a.WalletSign, mworker, r)
+	vrfout, err := ComputeVRF(ctx, a.WalletSign, mworker, miner, DSepElectionPost, r)
 	if err != nil {
 		return false, nil, xerrors.Errorf("failed to compute VRF: %w", err)
 	}
 
+	pset, err := a.StateMinerProvingSet(ctx, miner, ts)
+	if err != nil {
+		return false, nil, xerrors.Errorf("failed to load proving set for miner: %w", err)
+	}
+	log.Warningf("Proving set for miner %s: %s", miner, pset)
+
+	var sinfos []sectorbuilder.PublicSectorInfo
+	for _, s := range pset {
+		var commRa [32]byte
+		copy(commRa[:], s.CommR)
+		sinfos = append(sinfos, sectorbuilder.PublicSectorInfo{
+			SectorID: s.SectorID,
+			CommR:    commRa,
+		})
+	}
+	sectors := sectorbuilder.NewSortedPublicSectorInfo(sinfos)
+
+	hvrf := sha256.Sum256(vrfout)
+	log.Info("Replicas: ", sectors)
+	candidates, err := epp.GenerateCandidates(ctx, sectors, hvrf[:])
+	if err != nil {
+		return false, nil, xerrors.Errorf("failed to generate electionPoSt candidates: %w", err)
+	}
+
 	pow, err := a.StateMinerPower(ctx, miner, ts)
 	if err != nil {
 		return false, nil, xerrors.Errorf("failed to check power: %w", err)
 	}
 
-	return types.PowerCmp(vrfout, pow.MinerPower, pow.TotalPower), vrfout, nil
+	ssize, err := a.StateMinerSectorSize(ctx, miner, ts)
+	if err != nil {
+		return false, nil, xerrors.Errorf("failed to look up miners sector size: %w", err)
+	}
+
+	var winners []sectorbuilder.EPostCandidate
+	for _, c := range candidates {
+		if types.IsTicketWinner(c.PartialTicket[:], ssize, pow.TotalPower, 1) {
```

Comment:
Is this `1` the `challangeCount` in sectorbuilder? If yes, then it should be `1/25` of the proving set sectors (at least that's what I've heard)